### PR TITLE
355 mergecubes uses default geotrellis partitioner in certain cases which causes oom errors

### LINKED ISSUE #355: mergecubes uses default geotrellis partitioner in certain cases which causes OOM errors
The default SpacePartitioner uses 16x16 tiles per partition which requires too much memory if there are many bands, dates or if the tiles are large. This commit introduces a partitioner that attempts to control the partition size (max records) based on a maximum memory footprint. 

The new partitioner in this special case can have 1,2,4,8,16,32,64,128,256,512, or 1024 records in a single partition based on the dimensions of a single MultiBandTile: tile size, number of bands, and celltype.

### PR DESCRIPTION
**Pull request by JeroenVerstraelen**
The default SpacePartitioner uses 16x16 tiles per partition which requires too much memory if there are many bands, dates or if the tiles are large. This commit introduces a partitioner that attempts to control the partition size (max records) based on a maximum memory footprint.

The new partitioner in this special case can have 1,2,4,8,16,32,64,128,256,512, or 1024 records in a single partition based on the dimensions of a single MultiBandTile: tile size, number of bands, and celltype.

### CODE REVIEW

File: openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala [hunk 1]
```diff
@@ -708,14 +719,13 @@ class OpenEOProcesses extends Serializable {
         SpacePartitioner[K](kb)
       }
     } else {
+      // At least one partitioner is undefined.
       logger.info(s"Merging cubes with partitioners: ${leftCube.partitioner} - ${rightCube.partitioner} - many band case detected: $manyBands")
-      if(manyBands) {
```
Comment (jdries) [inline openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala:712]: why the immediate removal of manyBands?
I would prefer a more incremental change that takes things one step at a time, allowing to release this one faster without too much worries.

File: openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala [hunk 2]
```diff
@@ -708,14 +719,13 @@ class OpenEOProcesses extends Serializable {
...
-        val index: PartitionerIndex[K] = getManyBandsIndexGeneric[K]()
-        SpacePartitioner[K](kb)(implicitly,implicitly,index)
-      }else{
-        SpacePartitioner[K](kb)
-      }
-
+      val nrBands = leftCount.getOrElse(1) + rightCount.getOrElse(1)
+      val outputCellType = maybeCellType(leftCube).getOrElse(DoubleCellType).union(maybeCellType(rightCube).getOrElse(DoubleCellType))
+      val tileSize = maybeTileSize(leftCube).getOrElse(128)
```
Comment (jdries) [inline openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala]: this is wrong: geotrellis tilesize is actually returning width*height, so something needs to change

Comment (jdries) [reply openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala]: checking out usage of tilesize in getPartitionerIndexForMaxPartitionSize, the default should perhaps be 128*128
some scaladoc might also help to make sure that interpretation is clear

**Review by jdries — CHANGES_REQUESTED** [review]

File: openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala (hunk 1 above)
Comment (JeroenVerstraelen) [reply openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala:712]: ManyBands returned either ByTileSpatialPartitioner or SpaceTimeByMonthPartitioner which are both special cases of the configurable partitioners so I thought it was safe. But of course the new partitioner has to be more thoroughly tested to ensure it doesn't actually exceed its maximum allowed memory size so I will revert the manyBands case.

File: openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala (hunk 2 above)
Comment (JeroenVerstraelen) [reply openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala]: Pushed a fix.

File: openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala (hunk 1 above)
Comment (JeroenVerstraelen) [reply openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala:712]: Pushed a fix.

**Review by jdries — APPROVED** [review]
